### PR TITLE
Add Unit Test for FreeRTOS_Bitconfig

### PR DESCRIPTION
**Pull request by moninom1**
Description
-----------
As the FreeRTOS_Bitconfig was added for merging IPv4 and IPv6, this change adds the Unit test case for this new file along with 100% coverage.

Test Steps
-----------
Unit Test
```
cmake -S test/unit-test -B test/unit-test/build/ -DCMAKE_BUILD_TYPE=Debug
make -C test/unit-test/build/ all
```

Coverage Test
```
make -C ${BUILD_DIR} coverage
lcov --list --rc lcov_branch_coverage=1 ${BUILD_DIR}coverage.info
```

Checklist:
----------

- [x] I have tested my changes. No regression in existing tests.
- [x] I have modified and/or added unit-tests to cover the code changes in this Pull Request.

Related Issue
-----------

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/unit-test/FreeRTOS_BitConfig/FreeRTOS_BitConfig_utest.c [hunk 1]
```diff
@@ -0,0 +1,617 @@
...
+
+    xResult = xBitConfig_read_uc( &xConfig, pucData, SIZE_OF_BINARY_STREAM );
+
+    TEST_ASSERT_EQUAL( pdFALSE, xResult );
+}
+
+/**
+ * @brief This functions verifies failure of
+ *        reading from a bit-config struct with
+ *        incorrect size.
```
Comment (ActoryOu) [inline test/unit-test/FreeRTOS_BitConfig/FreeRTOS_BitConfig_utest.c:153]: What's "incorrect" size here? Can we use more specific word to describe?

File: test/unit-test/FreeRTOS_BitConfig/FreeRTOS_BitConfig_utest.c [hunk 2]
```diff
@@ -0,0 +1,617 @@
...
+}
+
+
+/**
+ * @brief This functions verifies failure to
+ *        Peek the last byte from a bit-config
+ *        struct as length check fails.
+ */
+
+void test_pucBitConfig_peek_last_index_uc_LenIncorrect( void )
```
Comment (ActoryOu) [inline test/unit-test/FreeRTOS_BitConfig/FreeRTOS_BitConfig_utest.c:272]: Same as above. More specific word instead of LenIncorrect.

File: test/unit-test/FreeRTOS_BitConfig/FreeRTOS_BitConfig_utest.c [hunk 3]
```diff
@@ -0,0 +1,617 @@
...
+    memset( &ucContents, 0, uxSize );
+    memset( ucContents, 1, uxSize );
+    pxConfig->xHasError = pdFALSE;
+    pxConfig->uxIndex = 0;
+    pxConfig->uxSize = uxSize;
+    pxConfig->ucContents = ucContents;
+
+    ucResult = ucBitConfig_read_8( pxConfig );
+
+    TEST_ASSERT_EQUAL( ucContents[ 0 ], ucResult );
```
Comment (ActoryOu) [inline test/unit-test/FreeRTOS_BitConfig/FreeRTOS_BitConfig_utest.c:356]:
```suggestion
    TEST_ASSERT_EQUAL( ucContents[ 0 ], ucResult );
    TEST_ASSERT_EQUAL( 1, pxConfig->uxIndex );
```

File: test/unit-test/FreeRTOS_BitConfig/FreeRTOS_BitConfig_utest.c [hunk 4]
```diff
@@ -0,0 +1,617 @@
...
+    pxConfig->ucContents = ucContents;
+
+    ulResultExpected = ( ( ( uint32_t ) ucContents[ 0 ] ) << 24 ) |
+                       ( ( ( uint32_t ) ucContents[ 1 ] ) << 16 ) |
+                       ( ( ( uint32_t ) ucContents[ 2 ] ) << 8 ) |
+                       ( ( ( uint32_t ) ucContents[ 3 ] ) );
+
+    ulResult = ulBitConfig_read_32( pxConfig );
+
+    TEST_ASSERT_EQUAL( ulResultExpected, ulResult );
```
Comment (ActoryOu) [inline test/unit-test/FreeRTOS_BitConfig/FreeRTOS_BitConfig_utest.c:450]: Please also check the move of index.

File: test/unit-test/FreeRTOS_BitConfig/FreeRTOS_BitConfig_utest.c [hunk 5]
```diff
@@ -0,0 +1,617 @@
...
+
+    vBitConfig_write_uc( &xConfig, pucData, SIZE_OF_BINARY_STREAM );
+}
+
+/**
+ * @brief This functions verifies failure to
+ *        writing any number of bytes from the bit stream.
+ */
+
+void test_vBitConfig_write_uc_IncorrectSize( void )
```
Comment (ActoryOu) [inline test/unit-test/FreeRTOS_BitConfig/FreeRTOS_BitConfig_utest.c:474]: No incorrect size.

File: test/unit-test/FreeRTOS_BitConfig/FreeRTOS_BitConfig_utest.c [hunk 6]
```diff
@@ -0,0 +1,617 @@
...
+    memset( &xConfig, 0, sizeof( BitConfig_t ) );
+    memset( ucContents, 0, uxSize );
+    xConfig.xHasError = pdFALSE;
+    xConfig.uxIndex = 0;
+    xConfig.uxSize = uxSize;
+    xConfig.ucContents = ucContents;
+
+    vBitConfig_write_8( &xConfig, ucValue );
+
+    TEST_ASSERT_EQUAL( sizeof( uint8_t ), xConfig.uxIndex );
```
Comment (ActoryOu) [inline test/unit-test/FreeRTOS_BitConfig/FreeRTOS_BitConfig_utest.c:533]: Check the content.

File: test/unit-test/FreeRTOS_BitConfig/FreeRTOS_BitConfig_utest.c [hunk 7]
```diff
@@ -0,0 +1,617 @@
...
+    memset( &xConfig, 0, sizeof( BitConfig_t ) );
+    memset( ucContents, 0, uxSize );
+    xConfig.xHasError = pdFALSE;
+    xConfig.uxIndex = 0;
+    xConfig.uxSize = uxSize;
+    xConfig.ucContents = ucContents;
+
+    vBitConfig_write_16( &xConfig, usValue );
+
+    TEST_ASSERT_EQUAL( sizeof( uint16_t ), xConfig.uxIndex );
```
Comment (ActoryOu) [inline test/unit-test/FreeRTOS_BitConfig/FreeRTOS_BitConfig_utest.c:556]: Check content

File: test/unit-test/FreeRTOS_BitConfig/FreeRTOS_BitConfig_utest.c [hunk 8]
```diff
@@ -0,0 +1,617 @@
...
+
+    memset( &xConfig, 0, sizeof( BitConfig_t ) );
+    xConfig.xHasError = pdFALSE;
+    xConfig.uxIndex = 0;
+    xConfig.uxSize = uxSize;
+    xConfig.ucContents = ucContents;
+
+    vBitConfig_write_32( &xConfig, usValue );
+
+    TEST_ASSERT_EQUAL( sizeof( uint32_t ), xConfig.uxIndex );
```
Comment (ActoryOu) [inline test/unit-test/FreeRTOS_BitConfig/FreeRTOS_BitConfig_utest.c:578]: Check content

File: test/unit-test/FreeRTOS_BitConfig/FreeRTOS_BitConfig_utest.c [hunk 9]
```diff
@@ -0,0 +1,617 @@
...
+ */
+void test_vBitConfig_releaseNULL( void )
+{
+    BitConfig_t xConfig, xConfigExpected;
+
+    memset( &xConfig, 0, sizeof( BitConfig_t ) );
+    memset( &xConfigExpected, 0, sizeof( BitConfig_t ) );
+    xConfig.ucContents = NULL;
+
+    vBitConfig_release( &xConfig );
```
Comment (ActoryOu) [inline test/unit-test/FreeRTOS_BitConfig/FreeRTOS_BitConfig_utest.c:593]: Can we add a test case for NULL pointer like below?
`vBitConfig_release( NULL );`

File: test/unit-test/FreeRTOS_BitConfig/FreeRTOS_BitConfig_utest.c [hunk 10]
```diff
@@ -0,0 +1,617 @@
...
+
+    TEST_ASSERT_EQUAL_MEMORY( &xConfigExpected, &xConfig, sizeof( BitConfig_t ) );
```
Comment (ActoryOu) [inline test/unit-test/FreeRTOS_BitConfig/FreeRTOS_BitConfig_utest.c:595]: What's this check for?
They're both set to 0. So they of course same, isn't it?
Do I miss anything?

File: test/unit-test/FreeRTOS_BitConfig/FreeRTOS_BitConfig_utest.c [hunk 11]
```diff
@@ -0,0 +1,617 @@
...
+    memset( &xConfig, 0, sizeof( BitConfig_t ) );
+    memset( ucContent, 1, SIZE_OF_BINARY_STREAM );
+    memset( &xConfigExpected, 0, sizeof( BitConfig_t ) );
+    xConfig.ucContents = ucContent;
+
+    vPortFree_ExpectAnyArgs();
+
+    vBitConfig_release( &xConfig );
+
+    TEST_ASSERT_EQUAL_MEMORY( &xConfigExpected, &xConfig, sizeof( BitConfig_t ) );
```
Comment (ActoryOu) [inline test/unit-test/FreeRTOS_BitConfig/FreeRTOS_BitConfig_utest.c:616]: Same as above

**Review by ActoryOu — CHANGES_REQUESTED** [review]

File: test/unit-test/FreeRTOS_BitConfig/FreeRTOS_BitConfig_utest.c (hunk 11 above)
Comment (moninom1) [reply test/unit-test/FreeRTOS_BitConfig/FreeRTOS_BitConfig_utest.c:616]: ucContent is set to 1s which should be cleared out.

Comment (moninom1) [reply test/unit-test/FreeRTOS_BitConfig/FreeRTOS_BitConfig_utest.c:616]: will update xConfig as well.

File: test/unit-test/FreeRTOS_BitConfig/FreeRTOS_BitConfig_utest.c (hunk 10 above)
Comment (moninom1) [reply test/unit-test/FreeRTOS_BitConfig/FreeRTOS_BitConfig_utest.c:595]: updated

File: test/unit-test/FreeRTOS_BitConfig/FreeRTOS_BitConfig_utest.c (hunk 6 above)
Comment (moninom1) [reply test/unit-test/FreeRTOS_BitConfig/FreeRTOS_BitConfig_utest.c:533]: done

**Review by ActoryOu — APPROVED** [review]

**Review by tony-josi-aws — APPROVED** [review]
